packages/x-components/src/x-modules/facets-next/service/facets.service.ts

```diff
@@ -46,6 +51,23 @@ export class BaseFacetsService implements FacetsService {
     });
   }
 
+  setFacets(facetGroup: FacetsGroup): void {
```

Comment:
Maybe we have to swap the names:

Update: updates the current filters with the new filters from API/Props (this keeps the current selected state)
Set: Sets the passed filters to the state (without changing the state, like any other setter)

Reply:
Also I think it's `setFacets(facetsGroup: FacetsGroup): void {`

Reply:
I get it wrong, it is already this way

Reply:
maybe the needed change is to rename `saveFilters` to `saveFiltersWithPreviousState` to keep it clearer.